src/base.iced

```diff
@@ -178,23 +178,38 @@ class Base
 
   #------
 
-  _v_include_pgp_hash : -> false
-  _v_require_full_pgp_hash : -> false
-  _v_pgp_km_to_hash : () -> @km()
-  _v_pgp_hash_dest : (body) -> body.key
+  _v_include_pgp_details : -> false
+  _v_require_pgp_details : -> false
+  _v_pgp_details_dest : (body) -> body.key
+  _v_pgp_km : () -> @km()
 
   full_pgp_hash : (opts, cb) ->
     if @_full_pgp_hash is undefined
       esc = make_esc cb
-      await @_v_pgp_km_to_hash()?.export_pgp_public {}, esc defer key
+      await @_v_pgp_km()?.export_pgp_public {}, esc defer key
       @_full_pgp_hash = if key? then hash_pgp_key key else null
     cb null, @_full_pgp_hash
 
-  check_pgp_hash: ({json}, cb) ->
+  _add_pgp_details : (ret, cb) ->
```

Comment:
I think `add_pgp_details` deserves a little bit of commentary saying what `ret` is. It's the objection within the final JSON blob that we're going to be modifying with the PGP details, but changes on the basis of the subclass, etc.  Should only be called with the output of `_v_pgp_details_dest`.  Or some such.